### ISSUE TITLE
Braiins OS answer/request

### ISSUE THREAD

**Issue by ErnesTRKS** [anselal/antminer-monitor#172]
Hi, One of my antminers S9 really needs this software but they no longer use Cgminer is there any way to configure this monitor to work with it?

**Comment by anselal** [comment]
Hi there,

sorry for the late reply. What do they use instead of cgminer ? cgminer is the default software for miners. It could be that the miner has just disabled access toe cgminer's API.

I need more info on this

**Comment by chainworksindustries** [comment]
If it is the Braiins OS as stated, it will use the new BOSMiner instead ...

Braiins (Slush Devs) have written it from scratch and it is open source also.

#crysx

**Comment by o2user** [comment]
Hi,

I second this feature :) Any attempts at implementing support?

https://braiins-os.com/plus

**Comment by anselal** [comment]
Difficult one. I got my hands in a S9 running BraiinsOS but there is no documentation for the API it uses.
It will take to much time to implement this...

**Comment by 3ehzad** [comment]
Braiins OS+ commands are almost the same + some new commands, BOSminere wrote from scratch in Rust language.
This is the API docs:
https://docs.braiins.com/os/plus-en/Development/1_api.html

**Comment by anselal** [comment]
> Braiins OS+ commands are almost the same + some new commands, BOSminer wrote from scratch in Rust language.
> This is the API docs:
> https://docs.braiins.com/os/plus-en/Development/1_api.html

you are right on that. will try to work on something but it will take some time cause the internet connection to my miner is reaaaally slow.

**Comment by anselal** [comment]
I managed to use the bosminer API to retrieve some info.
But the only available info is the Hashrate, Uptime and HW Error Rate %. There is no info about chip condition nor fan speed.

Maybe they are stil working on it ?

**Comment by 3ehzad** [comment]
All data + extra data are available this is the api link: 
IP.AD.RE.SS/cgi-bin/luci/admin/miner/api_status

you can use "?luci_username=root&luci_password=root" at the end of URL

**Comment by anselal** [comment]
I guess you are right. I didn't check all endpoints. 
I checked only stats, pools and summary, the defaults for cgminer.
Will look at it tomorrow.

Thnx 💪

**Comment by anselal** [comment]
@ErnesTRKS @chainworksindustries @o2user @3ehzad I managed to retrieve some info from the BOSminer API.
I will some changes to a new branch the next days so you can test it.

Sorry for the delay but I am currently on vacations :beach_umbrella:

**Comment by 3ehzad** [comment]
Please let me know if you need some tester :) @anselal

**Comment by papampi** [comment]
Any news on this?

**Comment by anselal** [comment]
> Any news on this?

Sorry for the late reply.
I had some miner issues and totally forgot about that.
Will have to recheck my code and try to apply all the changes
